src/snowflake/snowpark/mock/_functions.py

```diff
@@ -1161,9 +1161,84 @@ def nop(x):
         cast = cast_to_datetime
         sf_type = ts_type
     else:
-        raise ValueError(f"{part} is not a recognized date or time part.")
+        raise ValueError(f"{part} is not a recognized date or time part for dateadd.")
 
     res = datetime_expr.combine(
         value_expr, lambda date, duration: func(cast(date), duration)
     )
     return ColumnEmulator(res, sf_type=sf_type)
+
+
+@patch("date_part")
+def mock_date_part(part: str, datetime_expr: ColumnEmulator):
```

Comment:
we should have a separate JIRA for the usage notes on session parameter: https://docs.snowflake.com/en/sql-reference/functions/date_part#usage-notes